### PR TITLE
Added guidance on naming of model types

### PR DESCRIPTION
**Pull request by adrianhall**

### CODE REVIEW

File: docs/general/design.md [hunk 1]
```diff
@@ -81,6 +81,45 @@ The following are standard verb prefixes.  You should have a good (articulated)
...
+class ConfigurationSetting {
+    String key;
+    String value;
+    ConfigurationSettingDetails details;
+}
+{% endhighlight %}
+
+Optional parameters and settings to an operation should be collected into an options bag named `<operation>Options`. For example, the `GetConfigurationSetting` method might take a `GetConfigurationSettingOptions` class for specifying optional parameters.
+
+Results should use the model type (e.g. `ConfigurationSetting`) where the return value is a complete set of data for the model.  However, in cases where an incomplete set of data is returned, use the following types:
```
Comment (johanste) [inline docs/general/design.md]: How does this play with services that support $select/$expand type of projections? I would suggest that if the items in a list correspond to one of the supported shapes that a get would do (e.g. you can select only the name property in the get method, and list also only return the name property), then we can use the same type.

Comment (adrianhall) [reply docs/general/design.md]: Good question - that was not discussed in the board meeting - @KrzysztofCwalina @JonathanGiles @bterlson @JeffreyRichter - care to chime in?

Comment (JeffreyRichter) [reply docs/general/design.md]: I think what Johan is saying is that for projections that are a proper subset of an item, the item's type be used and the customer should know to avoid accessing properties/fields that they didn't have in the projection. I'm OK with this suggestion. In fact, I don't see any other possibility for type-safe languages.

Comment (johanste) [reply docs/general/design.md]: Then I would rephrase the "incomplete set of data" statement to "partial schema"

Comment (adrianhall) [reply docs/general/design.md]: Added.
